src/test/java/com/wb/test/WeatherForecastApiApplicationTests.java

```diff
@@ -0,0 +1,78 @@
+package com.wb.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import com.wb.controller.WeatherForecastController;
+import com.wb.service.WeatherService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.wb.payload.Weather;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class WeatherForecastApiApplicationTests {
+
+	@Mock
+	private WeatherService weatherService;
+	@Mock
+	Future<Weather> futureWeather = new CompletableFuture<Weather>();
+	@InjectMocks
+	WeatherForecastController controller = new WeatherForecastController();
```

Comment:
Używasz spring runnera czyli podnosisz caly kontekst springa. Spring tworzy ci wszystkie potrzebne obiekty a Ty go olewasz i tworzysz je samemu i wstrzykujesz mocki. Sa dwa podejscia, musisz wybrac 1 per klasa testowa.
1. Robisz springowy test(lacznie z jego kontekstem) i uzywasz @Autowired zeby uzyskac obiekty ktore chcesz przetestowac. To jest test bardziej integracyjny bo testujesz caly przekroj aplikacji. np controllera restowego.
2. Robisz test z uzyciem samego mockito i junit. Testujesz pojedyncze klasy -np Weather service. Wtedy uzywasz MockitoExtension + InjectMocks + Mock